internal/tpm/tpm_test.go

```diff
@@ -0,0 +1,123 @@
+package tpm
+
+import (
+	"crypto/rand"
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm/legacy/tpm2"
+	"github.com/stretchr/testify/require"
+)
+
+// These unit tests all use the tpm simulator from go-tpm-tools.
+
+type TestFixture struct {
+	tpm *TPM
+}
+
+func setupTestFixture(t *testing.T) *TestFixture {
+	t.Helper()
+
+	tpm, err := OpenTPMSimulator()
+	if err != nil {
```

Comment:
if this is an error wont tpm be nil I think you want this setup to return an error and you can handle error with testify after.

Reply:
Okay, trying that way